src/main/java/org/nuxeo/ecm/core/storage/marklogic/MarkLogicQueryBuilder.java

```diff
@@ -433,8 +436,8 @@ private QueryBuilder walkMixinTypes(List<Literal> mixins, boolean include) {
         // match on primary type
         QueryBuilder primaryQuery;
         FieldInfo primaryTypeInfo = walkReference(NXQL.ECM_PRIMARYTYPE);
-        if (primaryTypes.size() == 1 && rangeElementIndexes.stream().anyMatch(
-                new RangeElementIndexPredicate(NXQL.ECM_PRIMARYTYPE, "string"))) {
+        if (primaryTypes.size() == 1
+                && rangeElementIndexes.stream().anyMatch(new RangeElementIndexPredicate(KEY_PRIMARY_TYPE, "string"))) {
```

Comment:
You could put `new RangeElementIndexPredicate(KEY_PRIMARY_TYPE, "string")` in a constant, no need to build it every time.